pdfbox/__init__.py

```diff
@@ -176,3 +176,73 @@ def extract_text(self, input_path, output_path='',
+        """
+        Extract all text from PDF file.
+
+        Parameters
+        ----------
+        input_path : str
+            Input PDF file.
+        output_path : str
+            Output text file. If not specified, the extracted text is returned.
+        password : str
+            PDF password.
+        imageType : str
+            The image type to write to. Currently only jpg or png.
+            Default: jpg.
+        outputPrefix : str
+            The prefix to the image file.
+            DEfault: Name of PDF document.
+        startPage : bool
+            The first page to convert, one based.
+            Default: 1.
+        endPage : bool
+            The last page to convert, one based.
+            Default: Last Page.
+        page : int
+            The only page to extract (1-based).
+        dpi : int
+            DPI resolution of exported images.
+            Default: Detected from screen (or 96 if headless).
+        color : int
+            The color depth (valid: bilevel, gray, rgb, rgba).
+            Default: rgb.
+        cropbox : str
+            The page area to export.
+            Format: e.g "34 45 56 67"
+        time : int
+            Prints timing information to stdout.
+
+        Returns
+        -------
+        text : str
+            Time taken to complete the process.
+        """
+
+        options = (' -password {password}'.format(password=password) if password else '') + \
+                  (' -imageType {imageType}'.format(imageType=imageType) if imageType else '') + \
+                  (' -outputPrefix {outputPrefix}'.format(outputPrefix=outputPrefix) if outputPrefix else '') + \
+                  (' -startPage {startPage}'.format(startPage=startPage) if startPage else '') + \
+                  (' -endPage {endPage}'.format(endPage=endPage) if endPage else '') + \
+                  (' -page {page}'.format(page=page) if page else '') + \
+                  (' -dpi {dpi}'.format(dpi=dpi) if dpi else '') + \
+                  (' -color {color}'.format(color=color) if color else '') + \
+                  (' -cropbox {cropbox}'.format(cropbox=cropbox) if cropbox else '') + \
+                  (' -time {time}'.format(time=time) if time else '')
+
+        dst = os.path.join(output_path,os.path.basename(input_path))
+        dst = os.path.abspath(dst)
+        shutil.copy(input_path,dst)
+        cmd = '{java_path} -jar {pdfbox_path} PDFToImage {options} {input_path}'.format(java_path=self.java_path,
+                                                                                                       pdfbox_path=self.pdfbox_path,
+                                                                                                       options=options,
+                                                                                                       input_path=dst)
+        p = sarge.capture_stdout(cmd)
+        os.remove(dst)
+        if time:
+            return p.stdout.text
```

Comment:
Please add a unit test that can verify that your function works as expected; see the existing unit test to get an idea as to how to do so. I would create a 2-page test PDF and check that pdf_to_image generates two images as output.

Reply:
Sure, will do it.

Reply:
Have added the unittest, but have gone for a single page pdf.

Reply:
Adds two page pdf for unittest.
```
py -3.7 -m unittest tests/test_pdfbox.py
..
----------------------------------------------------------------------
Ran 2 tests in 4.115s

OK
```